demo/src/test/java/com/mongodb/jdbc/utils/TestUtils.java

```diff
@@ -0,0 +1,439 @@
+                m_isExecuting = false;
+            }
+
+            public boolean getIsExecuting()
+            {
+                return m_isExecuting;
+            }
+
+            public ResultSet getResultSet()
+            {
+                return m_result;
+            }
+
+            @Override
+            public void run()
+            {
+                try
+                {
+                    System.out.println("Executing '" + m_query + "'");
+                    m_isExecuting = true;
+                    m_result = m_statement.executeQuery(m_query);
+                    m_isExecuting = false;
+                }
+                catch (SQLException e)
+                {
+                    e.printStackTrace();
+                }
+            }
+        }
+
+        Statement statement = m_conn.createStatement();
+        ResultSet result = null;
+
+        QueryExecutor queryExecutor = new QueryExecutor(statement, query, result);
+        Thread execThread = new Thread(queryExecutor);
+        execThread.start();
+
+        while (!queryExecutor.getIsExecuting())
+        {
+            ;
+        }
+        if (execThread.isAlive())
+        {
+            try
+            {
+                // Still need to wait because the m_isExecuting flag is set
+                // before
+                // statement.executeQuery is called. Need to wait until an
+                // executor
+                // object is created.
+                Thread.sleep(2000);
+            }
+            catch (InterruptedException e)
+            {
+                e.printStackTrace();
+            }
+            System.out.println("Calling cancel");
+            statement.cancel();
+            System.out.println("The query cancelled successfully.");
+
+            // Try to use the statement again
+            queryExecutor = new QueryExecutor(statement, query, result);
+            execThread = new Thread(queryExecutor);
+            execThread.start();
+        }
+        else
+        {
+            System.out.println("The execution finished before cancel");
+            queryExecutor.getResultSet().close();
+        }
+
+        System.out
+                .println("-----------------------------------------------------------------------");
+    }
+
+    private void runExecuteQuery(String query, boolean printRs) throws Exception
+    {
+        try {
+            Statement stmt = m_conn.createStatement();
+            ResultSet rs = stmt.executeQuery(query);
+            if (printRs) {
+                PrintUtils.printResultSet(rs);
+            }
+            else {
+                PrintUtils.printResultSetMetadata(rs.getMetaData());
+            }
+        }
+        catch (Exception e)
+        {
+            e.printStackTrace();
+            throw e;
+        }
+    }
+
+    private void printRsInfo(ResultSet rs) throws Exception
+    {
+        try
+        {
+            //PrintUtils.printResultSetMetadata(rs.getMetaData());
```

Comment:
Similar to my comment in `TestUtils.java`, but was this accidentally commented out?